File: SEM_II/stack/stack.c
#include <stdio.h>
#include <stdlib.h>
#define size 5

struct stack
    {
        int store[size];
        int stacktop;
    };
    
void push(struct stack *s, int num);
int pop(struct stack *s);



void main()
{
    struct stack s;
    s.stacktop = -1;

    push(&s,16);
    printf("16 pushed...\n\n");

    printf("popped  %d",pop(&s));
}

void push(struct stack *s, int num)
{
    if(s->stacktop == size-1)
    {
        printf("OVERFLOW");
        exit(1);
    }
    else
    {
        s->stacktop++;
        s->store[s->stacktop] = num;
    }
}

int pop(struct stack *s)
{
    if (s->stacktop == -1)
    {
        printf("UNDERFLOW");
        exit(1);
    }
    else
    {
        int i = s->store[s->stacktop];
        s->stacktop--;
        return i;
    }
}

File: SEM_V/OOPS/stack/src/stack/Stack.java
package stack;

public class Stack<T> {
		private T[] items;
 		private int top;
 		
 		Stack(T[] arr){
 			items = arr;
 			top  = -1;
 		}
 			
		public void push(T x) {
			items[++top] = x;
			System.out.println("PUSHED: " + x );
		}
		
		public T pop() {
			if(!isEmpty()) {
				T popped = items[top--];

				System.out.println("POPPED: " + popped );
				return popped;
			}
			else 
				System.out.println("EMPTY STACK");
				return null;
		}
		
		public void display() {
			for (int i = 0; i<= top; i++) {
				System.out.println(items[i]);
			}
		}
		
		public int length() {
			System.out.printf("- LENGTH: %d\n" , top+1);
			return top +1;
		}
		
		public boolean isEmpty() {
			if(top == -1)
				return true;
			else 
				return false;
		}

}
